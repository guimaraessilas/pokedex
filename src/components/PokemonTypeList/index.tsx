import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { PokemonTypes } from '../../types/Pokemon';
import { colors } from '../../constants/colors';

type PokemonTypeProps = {
  types?: PokemonTypes[];
  horizontal?: boolean;
};

const PokemonTypeList = ({ types, horizontal }: PokemonTypeProps) => {
  return (
    <View style={[styles.container, horizontal && { flexDirection: 'row' }]}>
      {types?.map(({ type }, index) => (
        <View
          key={`${type.name}-${index}`}
          style={[
            styles.textContainer,
            { backgroundColor: `${colors[type?.name || 0]}` },
          ]}
        >
          <Text style={styles.typeTitle}>{type.name}</Text>
        </View>
      ))}
    </View>
  );
};

export default PokemonTypeList;

const styles = StyleSheet.create({
  textContainer: {
    height: 32,
    paddingHorizontal: 5,
    marginBottom: 4,
    marginRight: 12,
    borderRadius: 100,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
  },
  typeTitle: {
    padding: 5,
    color: 'white',
    fontSize: 12,
    fontWeight: '700',
    textTransform: 'capitalize',
  },
  container: {
    alignSelf: 'flex-start',
  },
});
