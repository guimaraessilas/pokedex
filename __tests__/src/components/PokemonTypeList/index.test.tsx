import React from 'react';
import 'react-native';
import { create } from 'react-test-renderer';
import PokemonTypeList from '../../../../src/components/PokemonTypeList';

const props = {
  types: [{ type: { name: 'grass' } }, { type: { name: 'poison' } }],
};

const setup = () => create(<PokemonTypeList {...props} />).toJSON();

it('Should render PokemonTypeList correctly', () => {
  const tree = setup();
  expect(tree).toMatchSnapshot();
});
