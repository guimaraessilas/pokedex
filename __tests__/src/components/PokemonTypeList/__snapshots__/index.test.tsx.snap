// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PokemonTypeList correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "flex-start",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 100,
          "flexDirection": "column",
          "height": 32,
          "justifyContent": "center",
          "marginBottom": 4,
          "marginRight": 12,
          "paddingHorizontal": 5,
        },
        {
          "backgroundColor": "#7AC74C",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "700",
          "padding": 5,
          "textTransform": "capitalize",
        }
      }
    >
      grass
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 100,
          "flexDirection": "column",
          "height": 32,
          "justifyContent": "center",
          "marginBottom": 4,
          "marginRight": 12,
          "paddingHorizontal": 5,
        },
        {
          "backgroundColor": "#A33EA1",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "700",
          "padding": 5,
          "textTransform": "capitalize",
        }
      }
    >
      poison
    </Text>
  </View>
</View>
`;
